tests/test_exceptions.py

```diff
@@ -50,13 +50,23 @@ def test_python_call_in_catch():
     assert d["good"] is True
 
 
+if hasattr(pytest, "PytestUnraisableExceptionWarning"):  # Python >= 3.8 and pytest >= 6
+    ignore_pytest_unraisable_warning = pytest.mark.filterwarnings(
+        "ignore::pytest.PytestUnraisableExceptionWarning"
+    )
+else:
+    ignore_pytest_unraisable_warning = lambda f: f  # NOQA: E731
+
+
+@ignore_pytest_unraisable_warning
 def test_python_alreadyset_in_destructor(monkeypatch, capsys):
```

Comment:
```python
def ignore_pytest_unraisable_warning():
    if hasattr(pytest, "PytestUnraisableExceptionWarning"):  # Python >= 3.8 and pytest >= 6
        return pytest.mark.filterwarnings("ignore::pytest.PytestUnraisableExceptionWarning")
    else:
        return lambda f: f

@ignore_pytest_unraisable_warning()
def test_python_alreadyset_in_destructor(monkeypatch, capsys):
```

That's if you want to be fancy.

Reply:
Does this have an advantage, moving things into a function?

Reply:
The only advantage, besides personal dislike for procedural global code, that I can think of is delayed evaluation of those 4 lines, which should speed up importing of the `test_exceptions` module. I'm not saying the difference would be significant.

Reply:
I'm _mildly_ in favor of this, but would be happy either way. Though the parens seem odd. Why not:

```python
def ignore_pytest_unraisable_warning(f):
    unraisable = "PytestUnraisableExceptionWarning"
    if hasattr(pytest, unraisable):  # Python >= 3.8 and pytest >= 6
        dec = pytest.mark.filterwarnings("ignore::pytest.{}".format(unraisable))
        return dec(f)
    else:
        return f
```

That's the blacked code.

Reply:
OK, yes, that looks better!

> Why not

Because I had decided the line wasn't too long, but black didn't agree and "fixed" it to this abomination.

Reply:
Thanks, @henryiii! That looks much better :-)

Reply:
If you have code like this:
```python
my_big_function_that_is_so_long("my_big_string_that_is_so_long")
```
And then black "fixes" it because it's too long, then write it like this:
```python
msg = "my_big_string_that_is_so_long"
my_big_function_that_is_so_long(msg)
```

Now black is happy, and it's more readable too. If you depend on the line length, say by tiling your monitor with windows, you don't want a random line poking off. And if you don't really mean the line length limit, change it - that's one of the (~2?) settings black allows. But I think this promotes better, more readable code (and I like 0 configuration - a so-so standard is better than perfect but non-standard).

Reply:
That's true. Let's just say I have not a lot of experience with forcibly keeping lines short. But this works, indeed.